### PR TITLE
[MM-23627] Fix deadlock on SendTypingEvent()

### PR DESCRIPTION
**Pull request by streamer45**
#### Summary

PR fixes yet another deadlock. Similarly to https://github.com/mattermost/mattermost-load-test-ng/pull/170, a failure to login would cause the ws client not to connect. When creating a post the user would still try to send the typing event regardless of the status of the connection. This would cause the user to block forever writing on the `wsTyping` channel.

PR also makes the `UserEntity.Connect()` method return an error to avoid a possible goroutine leak.

#### Ticket

https://mattermost.atlassian.net/browse/MM-23627

### CODE REVIEW

File: loadtest/control/simulcontroller/actions.go [hunk 1]
```diff
@@ -62,10 +68,13 @@ func (c *SimulController) reload(full bool) control.UserActionResponse {
 func (c *SimulController) login() control.UserActionResponse {
 	resp := control.Login(c.user)
 	if resp.Err != nil {
+		fmt.Println(resp.Err.Error())
```
Comment (isacikgoz) [inline loadtest/control/simulcontroller/actions.go]: Is this a debug line?

Comment (streamer45) [reply loadtest/control/simulcontroller/actions.go]: Indeed, good catch :+1:

**Comment by agnivade** [comment]
By goroutine leak, you are referring to this one right ?

```
go func() {
for err := range errChan {
			c.status <- c.newErrorStatus(err)
}
}()
```

So this will just be stuck reading errChan right ?

**Comment by streamer45** [comment]
> By goroutine leak, you are referring to this one right ?
> 
> ```
> go func() {
> for err := range errChan {
> 			c.status <- c.newErrorStatus(err)
> }
> }()
> ```
> 
> So this will just be stuck reading errChan right ?

Correct.

**Comment by agnivade** [comment]
Got it. So not technically a "leak" per se. But just a stuck goroutine.

**Comment by streamer45** [comment]
> Got it. So not technically a "leak" per se. But just a stuck goroutine.

It would be a goroutine that would never terminate until the program exited and you could have a new one every time the `Connect` failed, adding to the previous one.

**Comment by agnivade** [comment]
Ah I missed that. I thought no new ones would be created.

**Comment by streamer45** [comment]
> Ah I missed that. I thought no new ones would be created.

Yeah we create a new one every time the uses connects again, so every page reload or logout/login.

**Pull request merged**